src/cli/ask.rs

```diff
@@ -0,0 +1,131 @@
+use crate::cli::llm_client::LLMClients;
+use crate::cli::util::read_openai_api_key;
+use crate::state::State;
+use crate::OpenAIClient;
+use std::sync::{Arc, Mutex};
+use tokio::runtime::Runtime;
+use tokio::select;
+
+use crate::CtrlcFuture;
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, IntoPipelineData, PipelineData, ShellError, Signature, SyntaxShape, Value,
+};
+
+#[derive(Clone)]
+pub struct Ask {
+    state: Arc<Mutex<State>>,
+}
+
+impl Ask {
+    pub fn new(state: Arc<Mutex<State>>) -> Self {
+        Self { state }
+    }
+}
+
+impl Command for Ask {
+    fn name(&self) -> &str {
+        "ask"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("ask")
+            .required("question", SyntaxShape::String, "the question to be asked")
+            .optional(
+                "context",
+                SyntaxShape::Any,
+                "table of strings used as context for the question",
+            )
+            .category(Category::Custom("couchbase".to_string()))
+    }
+
+    fn usage(&self) -> &str {
+        "Asks a connected LLM a question, optionally enhanced with context"
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        ask(self.state.clone(), engine_state, stack, call, input)
+    }
+}
+
+pub fn ask(
+    state: Arc<Mutex<State>>,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    input: PipelineData,
+) -> Result<PipelineData, ShellError> {
+    let span = call.head;
+
+    let question: String = call.req(engine_state, stack, 0)?;
+    let mut context: Vec<String> = Vec::new();
+    match input.into_value(span) {
```

Comment:
I'm still a little dubious about us doing this everywhere, unless nushell also do it? Although it does work so...

Reply:
I agree, does feel a little hacky, and I can't see that nu-shell do it. But as you say it does work, and I'm not sure how else to achieve the same effect otherwise